pkg/systemd/hw-detect.es6

```diff
@@ -55,8 +55,49 @@ function findPCI(udevdb, info) {
     }
 }
 
+// Parse and add to info.dmi.dimm = [{"speed":"11 MHz","locator":"A3", ...}]
+function processDMIDE(res, info) {
+  var dimms = [];
+  var data = res.split("\n\n");
+  for (var i = 1; i < data.length - 1; i++) {
+      if (data[i].length == 0) {
+        continue;
+      }
+      var dict = {};
+      var speed = data[i].substring(data[i].indexOf("Speed: "), data[i].indexOf("Manufacturer") - 1);
+      var dimm_type = data[i].substring(data[i].indexOf("Type: "), data[i].indexOf("Type Detail:") - 1);
+      var size = data[i].substring(data[i].indexOf("Size: "), data[i].indexOf("Form Factor") - 1);
+      var type_detail = data[i].substring(data[i].indexOf("Type Detail: "), data[i].indexOf("Speed: ") - 1);
+      var locator = data[i].substring(data[i].indexOf("Locator: "), data[i].indexOf("Bank Locator:") - 1);
+      var serial = data[i].substring(data[i].indexOf("Serial Number: "), data[i].indexOf("Asset Tag: ") - 1);
+      var vendor = data[i].substring(data[i].indexOf("Manufacturer: "), data[i].indexOf("Serial Number: ") - 1);
+      var part_number = data[i].substring(data[i].indexOf("Part Number: "), data[i].indexOf("Rank: ") - 1);
+      if (speed.includes("Unknown") == true || speed == "") {
+          dict["locator"] = locator.split(":")[1].trim();
+          dict["size"] = "";
+          dict["type"] = "";
+          dict["type_detail"] = "[empty]";
```

Comment:
This needs to be marked as translatable. Also, not sure if the square brackets really add much meaning.